app/controllers/provider/passwords_controller.rb

```diff
@@ -48,10 +49,14 @@ def update
 
   private
 
+  def can_reset_password?
+    render_error 'Cannot reset password for Master Account. Please contact your system administrator.', status: 422 if domain_account.master?
+  end
+
   # Can't be used for neither Buyer nor Master
   #
   def find_provider
-    unless @provider ||= Account.providers_with_master.find_by_self_domain(request.host.to_s)
+    unless @provider ||= Account.providers.find_by_self_domain(request.host.to_s)
```

Comment:
2 things about this:

1. It is better to do `return if ...` and continue in the next line than doing a method where everything is nested inside a condition.
2. I prefer `Account.tenants` over `Account.providers`. Because `providers` sometimes means just tenants and sometimes means tenants + master. And look at this:
![image](https://user-images.githubusercontent.com/11318903/74697484-1dfd4480-51fb-11ea-97d9-cb5ac6be4aca.png)